source/fundamentals/gridfs.txt

```diff
@@ -0,0 +1,278 @@
+- :ref:`Retrieve File Information <gridfs-retrieve-file-info>`
+- :ref:`Download Files <gridfs-download-files>`
+- :ref:`Rename Files <gridfs-rename-files>`
+- :ref:`Delete Files <gridfs-delete-files>`
+- :ref:`Delete a GridFS Bucket <gridfs-delete-bucket>`
+
+How GridFS Works
+----------------
+
+GridFS organizes files in a **bucket**, which contains the following collections, named using the convention
+defined in the GridFS specification:
+
+- The ``chunks`` collection stores the binary file chunks.
+- The ``files`` collection stores the file metadata.
+
+When you create a new GridFS bucket, the driver creates the collections
+referenced above, prefixed with the default bucket name ``fs``, unless you specify a different name. The driver also creates an index on each
+collection to ensure efficient retrieval of files and related
+metadata. The driver only creates the GridFS bucket on the first write
+operation if it does not already exist. The driver only creates indexes if
+they do not exist and when the bucket is empty. For more information on
+GridFS indexes, see the server manual page on :manual:`GridFS Indexes </core/gridfs/#gridfs-indexes>`.
+
+When storing files with GridFS, the driver splits the files into smaller
+pieces, each represented by a separate document in the ``chunks`` collection.
+It also creates a document in the ``files`` collection that contains
+a file id, file name, and other file metadata. You can upload the file from
+memory or from a stream. The following diagram describes how GridFS splits
+files for storage in a bucket:
+
+.. figure:: /includes/figures/GridFS-upload.png
+   :alt: A diagram that shows how GridFS uploads a file to a bucket
+
+When retrieving files, GridFS fetches the metadata from the ``files``
+collection in the specified bucket and uses the information to reconstruct
+the file from documents in the ``chunks`` collection. You can read the file
+into memory or output it to a stream.
+
+.. _gridfs-create-bucket:
+
+Create a GridFS Bucket
+----------------------
+
+To store or retrieve files from GridFS, create a bucket or get a reference
+to an existing one on a MongoDB database. Create a ``GridFSBucket``
+instance, passing a ``db`` instance as the parameter. You can use the ``GridFSBucket`` instance to
+call read and write operations on the files in your bucket:
+
+.. code-block:: javascript
+
+   const db = client.db(dbName);
+   var bucket = new mongodb.GridFSBucket(db);
+
+To create or reference a bucket with a custom name other than the
+default name ``fs``, pass your bucket name as the second parameter to the
+create() method as shown below:
+
+.. code-block:: javascript
+
+   var bucket = new mongodb.GridFSBucket(db, { bucketName: 'myCustomBucket' });
+
+For more information, see the :node-api-4.0:`GridFSBucket API documentation <classes/gridfsbucket.html>`.
+
+.. _gridfs-upload-files:
+
+Upload Files
+-----------
+
+``GridFSBucket`` has an ``openUploadStream()`` method that creates an upload
```

Comment:
When talking about how to do something, it should be task oriented -- this means it should be in the format of: "Use _x,_ to do _y_".

Applies to all sections.

```suggestion
Use the ``openUploadStream()`` method to create an upload stream for a given file name .
```

Reply:
accepted